client/landing/gutenboarding/domain-suggestions/resolvers.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import { apiFetch } from '@wordpress/data-controls';
+
+/**
+ * Internal dependencies
+ */
+import { DomainSuggestionQuery } from './types';
+import { receiveDomainSuggestions } from './actions';
+
+export function* getDomainSuggestions( queryObject: DomainSuggestionQuery ) {
+	const url = 'https://public-api.wordpress.com/rest/v1.1/domains/suggestions';
+
+	// `credentials` and `mode` args are needed since we're accessing the WP.com REST API
+	// (rather than the WP Core REST API)
+	const suggestions = yield apiFetch( {
+		credentials: 'same-origin',
+		mode: 'cors',
+		url: addQueryArgs( url, queryObject ),
```

Comment:
Welp: Getting a type error (ts2345) for `queryObject`, complaining that

> The argument of type "DomainSuggestionQuery" cannot be assigned to the parameter of type ["InputArgsObject"](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/af97534c408862a31f485f9a7b89b92474c86812/types/wordpress__url/index.d.ts#L7-L9).
>  Index signature mssing in type "DomainSuggestionQuery"

Reply:
Fixed by adding an index signature in 2646c69468. Not super elegant, but hopefully okay.